src/api/app/models/user_ldap_strategy.rb

```diff
@@ -197,33 +197,46 @@ def self.find_with_ldap(login, password)
     # @@ldap_search_con.bound? doesn't catch it as well. So when an error occurs, we
     # simply it try it a seccond time, which forces the ldap connection to
     # reinitialize (@@ldap_search_con is unbound and nil).
+    ldap_first_try = true
     user = nil
     user_filter = ''
 
-    if @@ldap_search_con.nil?
-      @@ldap_search_con = initialize_ldap_con(CONFIG['ldap_search_user'], CONFIG['ldap_search_auth'])
-    end
-    ldap_con = @@ldap_search_con
-    if ldap_con.nil?
-      Rails.logger.info("Unable to connect to LDAP server")
-      return
-    end
+    1.times do
```

Comment:
Huh?

Reply:
Its how this was done before. Its a loop block (the redo line will redo the whole block). See:
https://github.com/openSUSE/open-build-service/commit/db44312746793b8dc46651af48de231f827d0cc6

Yes it looks weird and there is without a doubt better ways of doing this but since we are making a change to a "stable" branch 2.8 I decided to just revert things to what we know worked rather than find a more elegant solution.

Reply:
Please add at least a comment explaining why this is needed:-)

And thanks for fixing it!

Reply:
Comment added. Thanks